conan/cli/commands/test.py

```diff
@@ -55,12 +55,6 @@ def run_test(conan_api, path, ref, profile_host, profile_build, remotes, lockfil
                                                          update=update,
                                                          lockfile=lockfile,
                                                          tested_python_requires=tested_python_requires)
-    if isinstance(tested_python_requires, str):  # create python-require
-        conanfile = root_node.conanfile
-        if not getattr(conanfile, "python_requires", None) == "tested_reference_str":
```

Comment:
This warning was always failing, because ``python_requires`` in this stage is no longer a string, but an object.